### DISCUSSION TITLE
Can homebrew be proxied for internal network only devices?

### DISCUSSION THREAD

**Discussion by kitsushadow** [Homebrew/discussions#3116] [Everyday usage]
We are looking for ways to proxy the Homebrew packages, like Redhat Satellite, and also scan them with Vulnerability/Threat scanning tools. If things like this are already being done that would be helpful to understand.

**Comment by SMillerDev** [comment]
There are people mirroring Homebrew in china, looking into that might help you. There are no automated tools for scanning though.

**Comment by alebcay** [comment]
Per https://docs.brew.sh/Manpage#environment, you can set the following environment variables to tell Homebrew to get stuff from user-defined sources:
- `HOMEBREW_ARTIFACT_DOMAIN`: Prefix all download URLs, including those for bottles, with this value. For example, `HOMEBREW_ARTIFACT_DOMAIN=http://localhost:8080` will cause a formula with the URL `https://example.com/foo.tar.gz` to instead download from `http://localhost:8080/example.com/foo.tar.gz`.
- `HOMEBREW_BOTTLE_DOMAIN`: Use this URL as the download mirror for bottles. If bottles at that URL are temporarily unavailable, the default bottle domain will be used as a fallback mirror. For example, `HOMEBREW_BOTTLE_DOMAIN=http://localhost:8080` will cause all bottles to download from the prefix `http://localhost:8080/`. If bottles are not available at `HOMEBREW_BOTTLE_DOMAIN` they will be downloaded from the default bottle domain. _Default_: `https://ghcr.io/v2/homebrew/core`.
- `HOMEBREW_BREW_GIT_REMOTE`: Use this URL as the Homebrew/brew `git(1)` remote. _Default_: `https://github.com/Homebrew/brew`.
- `HOMEBREW_CORE_GIT_REMOTE`: Use this URL as the Homebrew/homebrew-core `git(1)` remote. _Default_: `https://github.com/Homebrew/homebrew-core`.

There are also some proxy-related environment variables that may be of interest to you:
- `all_proxy`: Use this SOCKS5 proxy for `curl(1)`, `git(1)` and `svn(1)` when downloading through Homebrew.
- `ftp_proxy`: Use this FTP proxy for `curl(1)`, `git(1)` and `svn(1)` when downloading through Homebrew.
- `http_proxy`: Use this HTTP proxy for `curl(1)`, `git(1)` and `svn(1)` when downloading through Homebrew.
- `https_proxy`: Use this HTTPS proxy for `curl(1)`, `git(1)` and `svn(1)` when downloading through Homebrew.
- `no_proxy`: A comma-separated list of hostnames and domain names excluded from proxying by `curl(1)`, `git(1)` and `svn(1)` when downloading through Homebrew.

[Example usage of these proxy-related environment variables is also documented](https://docs.brew.sh/Manpage#using-homebrew-behind-a-proxy).

**Reply by kitsushadow** [reply]
Thanks for that info, that is really helpful.
Where can i find the documentation for hosting my own homebrew? SMillerDev mentioned some folks in China are doing it i imagine someone within the staff knows what it takes to host the homebrew service?

**Reply by SMillerDev** [reply]
You can contact one of the mirrors, but they're not officially affeliated with homebrew so we don't really know how they run it.
